restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/client/api/ConferenceApiClient.java

```diff
@@ -0,0 +1,176 @@
+        super();
+        this.accountSid = accountSid;
+        this.name = name;
+        this.conferenceSid = conferenceSid;
+        this.storage = storage;
+    }
+    @Override
+    public void onReceive(Object message) throws Exception {
+        final Class<?> klass = message.getClass();
+        final ActorRef sender = sender();
+        ActorRef self = self();
+
+        if (logger.isInfoEnabled()) {
+            logger.info(" ********** ConferenceApiHelper " + self().path() + " Sender: " + sender);
+            logger.info(" ********** ConferenceApiHelper " + self().path() + " Processing Message: " + klass.getName());
+        }
+        if (StopConference.class.equals(klass)) {
+            onStopConference((StopConference) message, self, sender);
+        } else if (DownloaderResponse.class.equals(klass)) {
+            onDownloaderResponse(message, self, sender);
+        }
+    }
+
+    private void onDownloaderResponse(Object message, ActorRef self, ActorRef sender) {
+        final DownloaderResponse response = (DownloaderResponse) message;
+        if (logger.isInfoEnabled()) {
+            logger.info("Conference api response succeeded " + response.succeeded());
+            logger.info("Conference api response: " + response);
+        }
+        requestee.tell(message, self);
+    }
+    protected void onStopConference(StopConference message, ActorRef self, ActorRef sender) throws URISyntaxException, ParseException {
+        requestee = sender;
+        conferenceDetailRecord = getConferenceDetailRecord();
+        if(conferenceDetailRecord == null){
+            logger.error("could not retrieve conferenceDetailRecord");
+        } else {
+            try {
+                URI uri = new URI("/restcomm"+conferenceDetailRecord.getUri());
+                uri = UriUtils.resolve(uri);
+                if (logger.isInfoEnabled())
+                    logger.info("conference api uri is: "+uri);
+
+                Header[] headers = {
+                        new BasicHeader(HttpHeaders.AUTHORIZATION, getSuperAdminAuthenticationHeader())
+                        ,new BasicHeader("Content-type", "application/x-www-form-urlencoded")
+                        ,new BasicHeader("Accept", "application/json")
+                    };
+
+                ArrayList<NameValuePair> postParameters = new ArrayList<NameValuePair>();
+                postParameters.add(new BasicNameValuePair("Status", "Completed"));
+
+                httpAsycClientHelper = httpAsycClientHelper();
+                HttpRequestDescriptor httpRequestDescriptor =new HttpRequestDescriptor(uri, "POST", postParameters, -1, headers);
+                httpAsycClientHelper.tell(httpRequestDescriptor, self);
+            } catch (Exception e) {
+                logger.error("Exception while trying to terminate conference via api: ", e);
+            }
+        }
+    }
+
+    private String getSuperAdminAuthenticationHeader(){
```

Comment:
@maria-farooq I think we should use the credentials of the account that requested the stop conference initially and not the super admin.